src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/UpdateRecallComponentTest.kt

```diff
@@ -178,4 +184,53 @@ class UpdateRecallComponentTest : ComponentTestBase() {
       )
     )
   }
+
+  @Test
+  fun `updates a recall with "letter to prison" info`() {
+    val request = UpdateRecallRequest(
+      additionalLicenceConditions = true,
+      additionalLicenceConditionsDetail = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+      differentNomsNumber = false,
+      differentNomsNumberDetail = "Lorem ipsum dolor sit amet, consectetur adipiscing."
+    )
+    val response = authenticatedPatchRequest(recallPath, request)
+
+    assertThat(
+      response,
+      equalTo(
+        RecallResponse(
+          recallId = recallId,
+          nomsNumber = nomsNumber,
+          additionalLicenceConditions = request.additionalLicenceConditions,
+          additionalLicenceConditionsDetail = request.additionalLicenceConditionsDetail,
+          differentNomsNumber = request.differentNomsNumber,
+          differentNomsNumberDetail = request.differentNomsNumberDetail
+        )
+      )
+    )
+  }
+
+  @Suppress("unused")
+  private fun invalidLetterToPrisonFields(): Stream<String> {
+    return Stream.of(
+      """{"additionalLicenceConditions": "yes"}""",
+      """{"differentNomsNumber": "yes"}""",
+    )
+  }
+
+  @ParameterizedTest
+  @MethodSource("invalidLetterToPrisonFields")
+  fun `validates the 'letter to prison' boolean fields`(requestJson: String) {
```

Comment:
This looks like a test which can become completely generic?  Maybe a //TODO suggestion?